docs/operational_guide/bootstrapping.md

```diff
@@ -0,0 +1,106 @@
+
+| Node | Shard | State |
+|------|-------|-------|
+|  1   |   1   |   I   |
+|  1   |   2   |   I   |
+|  1   |   3   |   I   |
+|  2   |   1   |   A   |
+|  2   |   2   |   A   |
+|  2   |   3   |   A   |
+|  3   |   1   |   A   |
+|  3   |   2   |   A   |
+|  3   |   3   |   A   |
+
+Note that a bootstrap consistency level of majority is the default value, but can be modified by changing the value of the key "m3db.client.bootstrap-consistency-level" in EtcD to one of: "none", "one", "unstrict_majority" (attempt to read from majority, but settle for less if any errors occur), "majority" (strict majority), and "all".
+
+**Note**: Any bootstrappers configuration that does not include the peers bootstrapper will be unable to handle dynamic topology changes of any kind.
+
+### Uninitialized Topology Bootstrapper
+
+The purpose of the uninitialzied topology bootstrapper is to succeed bootstraps for all time ranges for shards that have never been completely bootstrapped. This allows us to run the default bootstrapper configuration of: filesystem,commitlog,peers,topology_uninitialized such that filesystem and commitlog are used by default in node restarts, peer bootstrapper is only used for node adds/removes/replaces, and bootstraps still succeed for brand new topologies where both the commitlog and peers bootstrappers will be unable to succeed any bootstraps. In other words, the uninitialized topology bootstrapper allows us to place the commitlog bootstrapper *before* the peers bootstrapper and still succeed bootstraps with brand new topologies without resorting to using the noop-all bootstrapper which suceeds bootstraps for all shard/time-ranges regardless of the status of the topology.
+
+The uninitialized topology bootstrapper determines whether a topology is "new" for a given shard by counting the number of hosts in the Initializing state and Leaving states and if the number of Initializing - Leaving > 0 than it succeeds the bootstrap because that means the topology has never reached a state where all hosts are Available.
+
+### No Operational All Bootstrapper
+
+The noop_all bootstrapper succeeds all bootstraps regardless of requests shards/time ranges.
+
+### Bootstrappers Configuration
+
+Now that we've gone over the various bootstrappers, lets consider how M3DB will behave in different configurations. Note that we include uninitialized_topology at the end of all the lists of bootstrappers because its required to get a new topology up and running in the first place, but is not required after that (although leaving it in has no detrimental effects). Also note that any configuration that does not include the peers bootstrapper will not be able to handle dynamic topology changes like node adds/removes/replaces.
+
+#### filesystem,commitlog,peers,uninitialized_topology (default)
```

Comment:
should these be represented as yaml arrays just as they are in the configuration?